java/com/facebook/crypto/streams/NativeMacLayeredOutputStream.java

```diff
@@ -34,14 +36,22 @@ public NativeMacLayeredOutputStream(NativeMac mac, OutputStream outputDelegate)
   @Override
   public void close() throws IOException {
     try {
+      appendMac();
+    } finally {
+      mOutputDelegate.close();
```

Comment:
So the semantics here seem to have changed, right? 

If you look at the original code, ```mOutputDelegate.close()``` was in the try block and here it's part of the finally. Try to fix that so the original behavior of the code doesn't change.